backend/src/main/java/com/soen341/instagram/service/impl/FollowingService.java

```diff
@@ -22,7 +24,14 @@
 	
 	public boolean isFollowing(String username)
 	{
-		return (accountRepository.doesUserFollow(UserAccessor.getCurrentAccount(accountRepository).getUsername(), username)==1);
+		if(!(SecurityContextHolder.getContext().getAuthentication() instanceof AnonymousAuthenticationToken))
+		{
```

Comment:
Can you leave a comment in the code as to why this is necessary?

Reply:
This is necessary because whenever a client is not logged in. You only want to show the following button as "Follow".